cdap-cli/src/main/java/co/cask/cdap/cli/CLIConnectionConfig.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli;
+
+import co.cask.cdap.client.config.ConnectionConfig;
+import co.cask.cdap.proto.Id;
+
+import java.net.URI;
+import javax.annotation.Nullable;
+
+/**
+ * {@link ConnectionConfig} that stores a username when available. Helps keep connectionConfig thread safe.
+ */
+public class CLIConnectionConfig extends ConnectionConfig {
+
+  private final Id.Namespace namespace;
+  private final String username;
+
+  public CLIConnectionConfig(ConnectionConfig connectionConfig, Id.Namespace namespace, @Nullable String username) {
+    super(connectionConfig.getHostname(), connectionConfig.getPort(), connectionConfig.isSSLEnabled());
```

Comment:
should we just add a ``ConnectionConfig`` constructor that accepts a ``ConnectionConfig``? would make this a little cleaner, perhaps?